Pegasus/include/geometry.hpp

```diff
@@ -0,0 +1,875 @@
+#ifndef PEGASUS_GEOMETRY_HPP
+#define PEGASUS_GEOMETRY_HPP
+
+#include <algorithm>
+#include <array>
+#include <functional>
+#include <limits>
+#include <memory>
+#include <cmath>
+#include <unordered_map>
+#include <vector>
+#include <utility>
+#include <cstdint>
+
+#include "Pegasus/include/math.hpp"
+
+namespace pegasus {
+namespace geometry {
+
+    //Shapes
+    class Shape {
+    public:
+        explicit Shape(Vector3 const & centerOfMass);
+
+        void setCenterOfMass(Vector3 const & centerOfMass);
+        Vector3 getCenterOfMass() const;
+
+    private:
+        Vector3 mCenterOfMass;
+    };
+
+    enum class SimpleShapeType : uint32_t {
+        PLANE,
+        TRIANGLE,
+        SPHERE,
+        CONE,
+        CYLINDER,
+        CAPSULE,
+        BOX,
+        NONE
+    };
+
+    class SimpleShape : public Shape {
+    public:
+        SimpleShapeType type;
+    public:
+        SimpleShape(Vector3 const & centerOfMass, SimpleShapeType type);
+    };
+
+    class Plane : public SimpleShape {
+    public:
+        Plane(Vector3 const & centerOfMass, Vector3 const & normal);
+
+        void setNormal(Vector3 const & normal);
+        Vector3 getNormal() const;
+
+    private:
+        Vector3 mNormal;
+    };
+
+    class Triangle : public SimpleShape {
+    public:
+        Triangle(Vector3 const & centerOfMass, Vector3 const & a, Vector3 const & b, Vector3 const & c);
+
+        void setAxes(Vector3 const & a, Vector3 const & b, Vector3 const & c);
+        void getAxes(Vector3& a, Vector3& b, Vector3& c) const;
+        Vector3 getNormal() const;
+
+    private:
+        Vector3 mA;
+        Vector3 mB;
+        Vector3 mC;
+        Vector3 mNormal;
+
+        void calculateNormal();
+    };
+
+    class Sphere : public SimpleShape {
+    public:
+        using Ptr = std::shared_ptr<Sphere>;
+
+    public:
+        Sphere(Vector3 const & centerOfMass, double r);
+
+        void setRadius(double r);
+        double getRadius() const;
+
+    private:
+        double mR;
+    };
+
+    class Cone : public SimpleShape {
+    public:
+        Cone(Vector3 const & centerOfMass, Vector3 const & a, double r);
+
+        void setAppex(Vector3 const & a);
+        Vector3 getAppex() const;
+
+        void setRadius(double r);
+        double getRadius() const;
+
+    private:
+        Vector3 mA;
+        double mR;
+    };
+
+    class Capsule : public SimpleShape {
+    public:
+        Capsule(Vector3 const & centerOfMass, Vector3 const & halfHeight, double r);
+
+        void setHalfHeight(Vector3 const & halfHeight);
+        Vector3 getHalfHeight() const;
+
+        void setRadius(double r);
+        double getRadius() const;
+
+    private:
+        Vector3 mHalfHeight;
+        double mR;
+    };
+
+    class Cylinder : public Capsule {
```

Comment:
Why not the other way around then? :D
Maybe either write two similar classes, or inherit both of them from new, more abstract class? Just an opinion, it's your call.